shared/composemanager/compose_manager.go

```diff
@@ -0,0 +1,170 @@
+package composemanager
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+
+	"github.com/chanzuckerberg/happy/shared/config"
+	"github.com/compose-spec/compose-go/loader"
+	"github.com/compose-spec/compose-go/types"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"gopkg.in/yaml.v2"
+)
+
+const (
+	platform_architecture = "platform_architecture"
+	build                 = "build"
+	services              = "services"
+)
+
+type ComposeManager struct {
+	HappyConfig *config.HappyConfig
+}
+
+func NewComposeManager() ComposeManager {
+	return ComposeManager{}
+}
+
+func (c ComposeManager) WithHappyConfig(happyConfig *config.HappyConfig) ComposeManager {
+	c.HappyConfig = happyConfig
+	return c
+}
+
+func (c ComposeManager) Generate(ctx context.Context) error {
+	p := types.Project{}
+	p.Services = types.Services{}
+
+	stackDef, err := c.HappyConfig.GetStackConfig()
+	if err != nil {
+		return errors.Wrap(err, "unable to get stack config")
+	}
+
+	_, ok := stackDef[services]
+	if !ok {
+		return errors.New("unable to find services in stack config")
+	}
+
+	servicesDef := stackDef[services].(map[string]any)
```

Comment:
not sure I understand the typecasting here. If you are going to typecase, handle the errors:

```suggestion
	servicesDef, ok := stackDef[services].(map[string]any)
	if !ok {
	  // return errors
	}
```